bundles/org.openhab.binding.emby/src/main/java/org/openhab/binding/emby/internal/handler/EmbyBridgeHandler.java

```diff
@@ -0,0 +1,307 @@
+        EmbyClientDiscoveryService discoveryService = this.clientDiscoveryService;
+        if ((discoveryService != null) && config.discovery) {
+            discoveryService.addDeviceIDDiscover(playstate);
+        }
+        this.getThing().getThings().forEach(thing -> {
+            EmbyDeviceHandler handler = (EmbyDeviceHandler) thing.getHandler();
+            if (handler != null) {
+                handler.handleEvent(playstate, hostname, embyport);
+                logger.trace("Handler was found for thing {}", thing.getLabel());
+            } else {
+                logger.trace("There is no handler for thing {}", thing.getLabel());
+            }
+        });
+    }
+
+    @Override
+    public void updateConnectionState(boolean connected) {
+        if (connected) {
+            updateStatus(ThingStatus.ONLINE);
+            reconnectionCount = 0;
+        } else {
+            reconnectionCount++;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "The connection to the emby server was closed, binding will wait one min and attempt to restablish connection. There have been "
+                            + Integer.toString(reconnectionCount) + " attempts to establish a new connection");
+            scheduler.schedule(() -> {
+                establishConnection();
+            }, 60000, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public void registerDeviceFoundListener(EmbyClientDiscoveryService embyClientDiscoverySerice) {
+        this.clientDiscoveryService = embyClientDiscoverySerice;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // 1) Handle REFRESH immediately on any bridge channel
+        if (command instanceof RefreshType) {
+            updateState(channelUID, pollCurrentValueBridge(channelUID));
+            return;
+        }
+
+        // 2) Everything else is read-only
+        logger.trace("Bridge received command '{}' on {}, but no write operations are supported here", command,
+                channelUID.getId());
+    }
+
+    /**
+     * Called by the discovery service to update the status string
+     * that backs the “discoveryStatus” channel.
+     */
+    public void updateDiscoveryStatus(String status) {
+        this.lastDiscoveryStatus = status;
+        // optionally push it immediately so rules/UI see it
+        updateState(new ChannelUID(getThing().getUID(), "discoveryStatus"), new StringType(status));
+    }
+
+    private EmbyBridgeConfiguration checkConfiguration() throws ConfigValidationException {
+        EmbyBridgeConfiguration embyConfig = getConfigAs(EmbyBridgeConfiguration.class);
+
+        if (embyConfig.api == null || embyConfig.api.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "There is no API key configured for this bridge, please add an API key to enable communication");
```

Comment:
1. Shorter is usually better. The user knowns the thing is offline due to a configuraiton error. The user only needs to know what parameter. 
2. Did you consider adding i18n custom keys to all thing state updates? It would enable translation of these detailed states. for reference: https://www.openhab.org/docs/developer/utils/i18n.html#using-custom-keys

Reply:
revised all thing status messages and updated to use i18n custom keys